### PR TITLE
Add ability to see open and closed issues at the same time

### PR DESCRIPTION
**Pull request by gwymor**
By clicking the currently active "Open" or "Closed" filter button in the
issue list, the user can toggle that filter off in order to see all
issues regardless of state. The URL "state" parameter will be set to
"all" and the "Open"/"Closed" button will not show as active.

### CODE REVIEW

**Comment by lunny** [comment]
Can you post some screenshots?

File: models/issues/tracked_time.go [hunk 1]
```diff
@@ -365,7 +365,10 @@ func getIssueTotalTrackedTimeChunk(ctx context.Context, opts *IssuesOptions, isC
...
 	}
 
-	return sumSession(opts, issueIDs).
-		And("issue.is_closed = ?", isClosed).
+	session := sumSession(opts, issueIDs)
+	if !isClosed.IsNone() {
+		session = session.And("issue.is_closed = ?", isClosed.IsTrue())
+	}
+	return session.
 		SumInt(new(trackedTime), "tracked_time.time")
```
Comment (denyskon) [inline models/issues/tracked_time.go]:
```suggestion
	return session.SumInt(new(trackedTime), "tracked_time.time")
```

File: routers/web/repo/issue.go [hunk 2]
```diff
@@ -237,10 +237,18 @@ func issues(ctx *context.Context, milestoneID, projectID int64, isPullOption uti
...
+	var isShowClosed util.OptionalBool
+	switch ctx.FormString("state") {
+	case "closed":
+		isShowClosed = util.OptionalBoolTrue
+	case "all":
+		isShowClosed = util.OptionalBoolNone
+	default:
+		isShowClosed = util.OptionalBoolFalse
+	}
+	// if there are closed issues and no open issues, default to showing closed
```
Comment (denyskon) [inline routers/web/repo/issue.go]:
```suggestion
	// if there are closed issues and no open issues, default to showing all issues
```

File: routers/web/repo/issue.go [hunk 3]
```diff
@@ -237,10 +237,18 @@ func issues(ctx *context.Context, milestoneID, projectID int64, isPullOption uti
...
 	if len(ctx.FormString("state")) == 0 && issueStats.OpenCount == 0 && issueStats.ClosedCount != 0 {
-		isShowClosed = true
+		isShowClosed = util.OptionalBoolTrue
```
Comment (denyskon) [inline routers/web/repo/issue.go]:
```suggestion
		isShowClosed = util.OptionalBoolNone
```

**Review by denyskon — APPROVED** [review]

File: routers/web/repo/issue.go (hunk 2 above)
Comment (gwymor) [reply routers/web/repo/issue.go]: Good idea, thanks!

**Review by lunny — APPROVED** [review]

**Pull request merged**